sentry-datasource-proxy/src/main/java/io/sentry/dsproxy/SentryQueryExecutionListener.java

```diff
@@ -0,0 +1,63 @@
+package io.sentry.dsproxy;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.HubAdapter;
+import io.sentry.IHub;
+import io.sentry.ISpan;
+import io.sentry.Span;
+import io.sentry.SpanStatus;
+import io.sentry.util.Objects;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.WeakHashMap;
+import java.util.stream.Collectors;
+import net.ttddyy.dsproxy.ExecutionInfo;
+import net.ttddyy.dsproxy.QueryInfo;
+import net.ttddyy.dsproxy.listener.QueryExecutionListener;
+import org.jetbrains.annotations.NotNull;
+
+/** datasource-proxy query execution listener that creates {@link Span}s around database queries. */
+@Open
+public class SentryQueryExecutionListener implements QueryExecutionListener {
+  private final @NotNull IHub hub;
+  private final @NotNull Map<String, ISpan> spans =
+      Collections.synchronizedMap(new WeakHashMap<>());
+
+  public SentryQueryExecutionListener(final @NotNull IHub hub) {
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+  }
+
+  public SentryQueryExecutionListener() {
+    this(HubAdapter.getInstance());
+  }
+
+  @Override
+  public void beforeQuery(
+      final @NotNull ExecutionInfo execInfo, final @NotNull List<QueryInfo> queryInfoList) {
+    final ISpan parent = hub.getSpan();
```

Comment:
here we assume that the transaction is always started by the eg spring boot auto instrumentation right? so those listeners should be always used along with other auto instrumentation or manual instrumentation but not standalone right, it'd be nice to add this to our docs if not yet

Reply:
Short note added: https://github.com/getsentry/sentry-docs/pull/3353/commits/b6bf4c860ef67a13825881a086f2647436bdf713
I think at this point we can assume that users know what a "Sentry transaction" is.